### PR TITLE
Feat/production apk

### LINKED ISSUE #5728: suite-native: Strategy for mobile app GA - de-Googled Android phones
We have prepared a distribution strategy for Android users using google services (with Firebase App Distribution and Google Play Store - https://github.com/trezor/trezor-suite/issues/5429). 

There should be also an option to distribute the Android app to users who are complete without Google services.

Proposals:
- Amazon (https://www.amazon.com/gp/mas/get/amazonapp)
- Aptoide (https://en.aptoide.com/)
- APK Mirror (https://www.apkmirror.com/apk/meizu/app-store/)
- APKPure (https://m.apkpure.com/)

Sources:
- https://www.cnet.com/tech/mobile/how-to-install-apps-outside-of-google-play/#:~:text=From%20your%20smartphone%20or%20tablet,warned%20before%20installing%20harmful%20apps.
- https://www.alphr.com/download-apps-android-without-google-play/

### PR DESCRIPTION
**Pull request by juriczech**
## Description
Shell script for uploading android production APK to newly created GH release draft. This will allow us to distribute APK without the official stores and users can download new versions from here.

Tested installation of APK downloaded from GH on Samsung Android and works as expected.

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/5728

## Screenshots:
Example of how the newly created draft will look like:
image

### CODE REVIEW

File: suite-native/app/eas-post-success.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    # Upload production APK to newly created GitHub release
+    curl -L \
+        -X POST \
+        -H "Accept: application/vnd.github+json" \
+        -H "Authorization: Bearer $GITHUB_TREZOR_SUITE_TOKEN" \
+        -H "X-GitHub-Api-Version: 2022-11-28" \
+        -H "Content-Type: application/octet-stream" \
+        "${upload_url}" \
+        --data-binary "@$EAS_BUILD_WORKINGDIR/suite-native/app/android/app/build/outputs/apk/release/app-release.apk" > /tmp/asset_response.json
```
Comment (PeKne) [inline suite-native/app/eas-post-success.sh:53]: Would be nice to rename the `.apk` so its reflects the version of the app.

Something like:
`apk-release.apk` --> `Trezor-Suite-Lite-24.4.1.apk` same as [desktop](https://github.com/trezor/trezor-suite/releases/tag/v24.3.4) does.

File: .github/workflows/suite-native_production_ci.yml [hunk 2]
```diff
@@ -75,3 +76,31 @@ jobs:
           --auto-submit
           --message ${{ github.sha }}
         working-directory: suite-native/app
+  androidAPK:
```
Comment (PeKne) [inline .github/workflows/suite-native_production_ci.yml:79]: Why does the app have to be build separately for GH release? Could not we use the already built `.apk` from `android` job of this file?

Comment (juriczech) [reply .github/workflows/suite-native_production_ci.yml:79]: Android production job outputs AAB file that we would need to convert to the APK to be uploaded. It's unnecessary complication in my opinion and would have to rely on some converter. This way you just build the APK separately and upload it.

File: suite-native/app/eas-post-success.sh (hunk 1 above)
Comment (juriczech) [reply suite-native/app/eas-post-success.sh:53]: Good catch, thanks. I had it in my forked version and forgot to update here. 

However this is just a path where to take the built APK from on the eas servers. We just need to change the name param in GH rest API, see https://github.com/trezor/trezor-suite/pull/11865/commits/2e88bd0d5496e509c49141b87ae313414671afc7

File: .github/workflows/suite-native_production_ci.yml (hunk 2 above)
Comment (PeKne) [reply .github/workflows/suite-native_production_ci.yml:79]: Got it 👍

**Review by PeKne — APPROVED** [review]

**Review by vdovhanych — APPROVED** [review]

**Review by matejkriz — APPROVED** [review]
🎉

**Pull request merged**